lib/galaxy/web/framework/helpers/grids.py

```diff
@@ -285,31 +286,146 @@ def url( *args, **kwargs ):
+                'extra'             : extra
+            })
+        for operation in self.operations:
+            grid_config['operations'].append({
+                'allow_multiple'        : operation.allow_multiple,
+                'allow_popup'           : operation.allow_popup,
+                'target'                : operation.target,
+                'label'                 : operation.label,
+                'confirm'               : operation.confirm,
+                'global_operation'      : False
+            })
+            if operation.allow_multiple:
+                grid_config['show_item_checkboxes'] = True
+            if operation.global_operation:
+                grid_config['global_operation'] = url( ** (operation.global_operation()) )
+        for action in self.global_actions:
+            grid_config['global_actions'].append({
+                'url_args'  : url(**action.url_args),
+                'label'     : action.label,
+                'target'    : action.target
+            })
+        for operation in [op for op in self.operations if op.async_compatible]:
+            grid_config['async_ops'].append(operation.label.lower())
+        for column in self.columns:
+            if column.filterable is not None and not isinstance( column, TextColumn ):
+                grid_config['categorical_filters'][column.key] = dict([ (filter.label, filter.args) for filter in column.get_accepted_filters() ])
+        for i, item in enumerate( query ):
+            item_dict = {
+                'id'                    : item.id,
+                'encode_id'             : trans.security.encode_id(item.id),
+                'link'                  : [],
+                'operation_config'      : {},
+                'column_config'         : {}
+            }
+            for column in self.columns:
+                if column.visible:
+                    link = column.get_link(trans, self, item)
+                    if link:
+                        link = url(**link)
+                    else:
+                        link = None
+                    target = column.target
+                    value = column.get_value( trans, self, item )
+                    if isinstance(value, str):
+                        value = unicode(value, 'utf-8')
+                        value = value.replace('/', '//')
+                    item_dict['column_config'][column.label] = {
+                        'link'      : link,
+                        'value'     : value,
+                        'target'    : target
+                    }
+            for operation in self.operations:
+                item_dict['operation_config'][operation.label] = {
+                    'allowed'   : operation.allowed(item),
+                    'url_args'  : url( **operation.get_url_args( item ) )
+                }
+            grid_config['items'].append(item_dict)
         trans.log_action( trans.get_user(), text_type( "grid.view" ), context, params )
```

Comment:
This log statement only happens for new-style (dict_format) now -- it should probably happen in both branches.